cmd/tls_test.go

```diff
@@ -0,0 +1,100 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	"github.com/arschles/assert"
+	"github.com/deis/controller-sdk-go/api"
+	"github.com/deis/workflow-cli/pkg/testutil"
+)
+
+func TestTLSInfo(t *testing.T) {
+	t.Parallel()
+	cf, server, err := testutil.NewTestServerAndClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer server.Close()
+	var b bytes.Buffer
+	cmdr := DeisCmd{WOut: &b, ConfigFile: cf}
+
+	server.Mux.HandleFunc("/v2/apps/numenor/tls/", func(w http.ResponseWriter, r *http.Request) {
+		testutil.SetHeaders(w)
+		fmt.Fprintf(w, `{
+	"uuid": "c4aed81c-d1ca-4ff1-ab89-d2151264e1a3",
+	"app": "numenor",
+	"owner": "nazgul",
+	"created": "2016-08-22T17:40:16Z",
+	"updated": "2016-08-22T17:40:16Z",
+	"https_enforced": true
+}`)
+	})
+
+	err = cmdr.TLSInfo("numenor")
+	assert.NoErr(t, err)
+	assert.Equal(t, b.String(), `=== numenor TLS
+HTTPS Enforced: true
+`, "output")
+}
+
+func TestTLSEnable(t *testing.T) {
+	t.Parallel()
+	cf, server, err := testutil.NewTestServerAndClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer server.Close()
+	var b bytes.Buffer
+	cmdr := DeisCmd{WOut: &b, ConfigFile: cf}
+
+	server.Mux.HandleFunc("/v2/apps/numenor/tls/", func(w http.ResponseWriter, r *http.Request) {
+		testutil.SetHeaders(w)
+		testutil.AssertBody(t, []byte(`{"https_enabled": true}`), r)
```

Comment:
The reason I did not go with that function call is because we'd have to do the following: https://github.com/deis/controller-sdk-go/blob/master/tls/tls_test.go#L171-L179

`api.NewTLS()` returns a TLS struct with HTTPSEnabled set to `false`. I went with the lazy approach to check that the response body is what we'd expect rather than set up a TLS object and test against that. Less code overall.

If you'd like me to implement the change then I'll change it, but the end result is still the same: we are asserting that the response body returns a bytestring with `{"https_enabled": true}` as the content.